Byte Buddy version update

Continuing the investigation.

```
Caused by: java.lang.NegativeArraySizeException: -1
	at tech.httptoolkit.relocated.net.bytebuddy.utility.dispatcher.JavaDispatcher$DynamicClassLoader.invoker(SourceFile:34)
	at tech.httptoolkit.relocated.net.bytebuddy.utility.dispatcher.JavaDispatcher$Invoker$CreationAction.run(SourceFile:1)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at tech.httptoolkit.relocated.net.bytebuddy.utility.dispatcher.JavaDispatcher.<clinit>(SourceFile:12)
```
This exception starts from version [1.11.6](https://github.com/raphw/byte-buddy/releases/tag/byte-buddy-1.11.6). By looking at the generated source mapping I see that the attempt to create array with negative size is at this line:
https://github.com/raphw/byte-buddy/blob/c51396ba504c0a7397d912f8e621d060bc360483/byte-buddy-dep/src/main/java/net/bytebuddy/utility/dispatcher/JavaDispatcher.java#L1270

Which fails as in the byte code `Invoker` interface was replaced with concrete class (in this commit it's `Unsafe` which is a class and has hashCode which has zero arguments... so my **guess** it's that it fails on this.

-- fast forward to latest byte buddy

Return value of JavaDispatcher.invoker() is [`net.bytebuddy.utility.Invoker`](https://github.com/raphw/byte-buddy/blob/master/byte-buddy-dep/src/main/java/net/bytebuddy/utility/Invoker.java) interface but in the byte code (after r8) it is substituted with [`DirectInvoker` ](https://github.com/raphw/byte-buddy/blob/cbcee69074d6401ce4a2bd193d42fe64c8bec51a/byte-buddy-dep/src/main/java/net/bytebuddy/utility/dispatcher/JavaDispatcher.java#L455) class which again has one method with 0 arguments.

The exception is thrown at this line:
https://github.com/raphw/byte-buddy/blob/master/byte-buddy-dep/src/main/java/net/bytebuddy/utility/dispatcher/JavaDispatcher.java#L1287

This can be fixed by leaving the `Invoker` interface untouched by adding
```
-keep class tech.httptoolkit.relocated.net.bytebuddy.utility.Invoker { *; }
```
to `r8-rules.txt`. This way the interface is not replaced and byte buddy is able to create DynamicClassloader.

Afterwards the agent is able to attach to processes like  before and we reach the second (or actually it is the first issue). 

Since version [1.11.3](https://github.com/raphw/byte-buddy/releases/tag/byte-buddy-1.11.3) we are greeted by multiple SSLHandshakeExceptions during tests:
```
Unexpected failure for new http-url-conn: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```
I currently have no idea what caused this... yet... 

---

Looking at the standalone test the exception is also there (but it manages to succeed?)
https://github.com/httptoolkit/jvm-http-proxy-agent/runs/4601280285?check_suite_focus=true

--- a/build.gradle
+++ b/build.gradle
@@ -29,7 +29,7 @@ configurations {
 }
 
 dependencies {
-    implementation group: 'net.bytebuddy', name: 'byte-buddy-dep', version: '1.11.2'
+    implementation group: 'net.bytebuddy', name: 'byte-buddy-dep', version: '1.12.6'
     // byte buddy contains references to jna classes and without them the r8Jar step fails
     compileOnly group: 'net.java.dev.jna', name: 'jna', version: '5.8.0'
 

--- a/build.gradle
+++ b/build.gradle
@@ -143,11 +143,13 @@ task distJar(type: Jar) {
     // Pull raw advice classes from the shadow JAR, unminified:
     from (zipTree(shadowJar.getArchiveFile())) {
         include "tech/httptoolkit/javaagent/advice/**/*"
+        include "tech/httptoolkit/relocated/net/bytebuddy/agent/builder/**/*"
     }
 
     // Pull other source & bundled dependencies in their minified form, from R8:
     from (zipTree(r8Jar.outputs.files[0])) {
         exclude "tech/httptoolkit/javaagent/advice/**/*"
+        exclude "tech/httptoolkit/relocated/net/bytebuddy/agent/builder/**/*"
     }
 }
 

--- a/r8-rules.txt
+++ b/r8-rules.txt
@@ -1,8 +1,9 @@
 -dontobfuscate
+-dontoptimize
 -allowaccessmodification
 -keepattributes SourceFile, LineNumberTable, *Annotation*
 
--keep class tech.httptoolkit.relocated.net.bytebuddy.utility.Invoker { *; }
+-keep class tech.httptoolkit.relocated.net.bytebuddy.agent.builder.** { *; }
 
 -keep class tech.httptoolkit.javaagent.** { *; }
 -keep class tech.httptoolkit.relocated.net.bytebuddy.asm.** { *; }

--- a/r8-rules.txt
+++ b/r8-rules.txt
@@ -2,5 +2,7 @@
 -allowaccessmodification
 -keepattributes SourceFile, LineNumberTable, *Annotation*
 
+-keep class tech.httptoolkit.relocated.net.bytebuddy.utility.Invoker { *; }
+
 -keep class tech.httptoolkit.javaagent.** { *; }
 -keep class tech.httptoolkit.relocated.net.bytebuddy.asm.** { *; }

